lib/Parser/Parser.cpp

```diff
@@ -712,6 +712,54 @@ bool Parser::parseLine(std::string &ErrStr) {
         if (!LHS)
           return false;
 
+        llvm::APInt DemandedBitsVal = APInt(LHS->Width, 0, false);
+        llvm::APInt ConstOne(LHS->Width, 1, false);
+        if (CurTok.K == Token::OpenParen) {
+          if (!consumeToken(ErrStr))
+            return false;
+          if (CurTok.K != Token::Ident) {
+            ErrStr = makeErrStr("invalid demandedBits token type");
+            return false;
+          }
+          if (CurTok.str() != "demandedBits") {
+            ErrStr = makeErrStr("invalid demandedBits data flow fact");
+            return false;
+          }
+          if (!consumeToken(ErrStr))
+            return false;
+          if (CurTok.K != Token::Eq) {
+            ErrStr = makeErrStr("exoected '=' for demandedBits");
+            return false;
+          }
+          if (!consumeToken(ErrStr))
+            return false;
+          if (CurTok.K != Token::UntypedInt) {
+            ErrStr = makeErrStr("expected demandedBits pattern of type [0|1]+");
+            return false;
+          }
+          if (LHS->Width != CurTok.Len) {
+            ErrStr = makeErrStr("demandedBits pattern must be of same length as infer operand width");
+            return false;
+          }
+          std::string DemandedBitsPattern = CurTok.str();
+          for (unsigned i=0; i<LHS->Width; ++i) {
+            if (DemandedBitsPattern[i] == '1')
+              DemandedBitsVal += ConstOne.shl(DemandedBitsPattern.length() - 1 - i);
```

Comment:
please put curly braces around the if-block if you put them around the else-block